Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs

```diff
@@ -427,11 +427,6 @@ private void CleanupObservedObjects()
         {
             if (Application.isPlaying)
             {
-                // Cleanup the scene objects are managing
-                if (observedObjectParent != null)
-                {
-                    observedObjectParent.transform.DetachChildren();
```

Comment:
@keveleigh I checked all references to observedObjectParent and everything that's ever being childed to it and could not find a reason why the detach is performed prior to deleting all of those.
At the point where meshes are added as children, they are also added to the list that's being cleared after the detaching.
I even manually called this function and logged child and list count and they were equal.